src/Components/Components/src/DefaultComponentActivator.cs

```diff
@@ -10,10 +11,11 @@ internal class DefaultComponentActivator : IComponentActivator
         public static IComponentActivator Instance { get; } = new DefaultComponentActivator();
 
         /// <inheritdoc />
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2067", Justification = "Requires a gesture that ensures components are always preserved. https://github.com/mono/linker/issues/1806")]
         public IComponent CreateInstance(Type componentType)
```

Comment:
You should flow `DynamicallyAccessedMembers.ParameterlessConstructor` annotation to this type. And if necessary, suppress it upstream.